File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) {
        int low = Integer.MAX_VALUE, max = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < low) {
                low = prices[i];
            }
            max = Math.max(max, prices[i] - low);
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
      s = set(nums)
      m = 0
      for i in nums:
        curr = 1
        if i - 1 not in s:
            while i + 1 in s:
               curr += 1
               i += 1
        m = max(curr, m)
      return m
       
